components/webdriver_server/lib.rs

```diff
@@ -1066,43 +1066,33 @@ impl Handler {
 
     /// <https://w3c.github.io/webdriver/#get-window-handles>
     fn handle_window_handles(&mut self) -> WebDriverResult<WebDriverResponse> {
-        let mut handles = self.get_window_handles()?;
+        let mut handles = self.get_window_handles();
         handles.sort();
 
         Ok(WebDriverResponse::Generic(ValueResponse(
             serde_json::to_value(handles)?,
         )))
     }
 
-    fn get_window_handle(&mut self, webview_id: WebViewId) -> WebDriverResult<String> {
-        self.get_window_handles()?
+    fn get_window_handle(&mut self, webview_id: WebViewId) -> Option<String> {
+        self.get_window_handles()
             .iter()
             .find(|id| id == &&webview_id.to_string())
             .cloned()
-            .ok_or_else(|| {
-                WebDriverError::new(
-                    ErrorStatus::UnknownError,
-                    "No such window while getting window handle",
-                )
-            })
     }
 
-    fn get_window_handles(&self) -> WebDriverResult<Vec<String>> {
-        let handles = self
-            .get_all_webview_ids()?
+    fn get_window_handles(&self) -> Vec<String> {
+        self.get_all_webview_ids()
             .into_iter()
             .map(|id| id.to_string())
-            .collect::<Vec<_>>();
-
-        Ok(handles)
+            .collect::<Vec<_>>()
```

Comment:
The compiler should be able to figure out what type is required here by type inference:

```suggestion
            .collect()
```